components/brave_new_tab_ui/reducers/rewards_reducer.ts

```diff
@@ -0,0 +1,189 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { Reducer } from 'redux'
+
+// Constants
+import { types } from '../constants/new_tab_types'
+
+// API
+import { InitialRewardsData, PreInitialRewardsData } from '../api/initialData'
+import * as storage from '../storage'
+import { getTotalContributions } from '../rewards-utils'
+
+const initialState = storage.load()
+
+export const rewardsReducer: Reducer<NewTab.State | undefined> = (state: NewTab.State | undefined, action: any) => {
+  if (state === undefined) {
+    state = initialState
+  }
+
+  const startingState = state
+  const payload = action.payload
+  switch (action.type) {
+    case types.CREATE_WALLET:
+      chrome.braveRewards.createWallet()
+      state = { ...state }
+      state.rewardsState.walletCreating = true
+      break
+
+    case types.ON_ENABLED_MAIN:
+      state = { ...state }
+      state.rewardsState.enabledMain = payload.enabledMain
+      break
+
+    case types.ON_WALLET_INITIALIZED: {
+      const result: NewTab.RewardsResult = payload.result
+      state = { ...state }
+
+      switch (result) {
+        case NewTab.RewardsResult.WALLET_CORRUPT:
+          state.rewardsState.walletCorrupted = true
+          break
+        case NewTab.RewardsResult.WALLET_CREATED:
+          state.rewardsState.walletCreated = true
+          state.rewardsState.walletCreateFailed = false
+          state.rewardsState.walletCreating = false
+          state.rewardsState.walletCorrupted = false
+          chrome.braveRewards.saveAdsSetting('adsEnabled', 'true')
```

Comment:
What's happening here? Are we using the NTP to perform some rewards service flow?

Reply:
At all points of opt-in for the first time we also set ads to be enabled:

https://github.com/brave/brave-core/blob/master/components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts#L37
https://github.com/brave/brave-core/blob/master/components/brave_rewards/resources/page/reducers/wallet_reducer.ts#L33

Reply:
Is there a reason that kind of business logic isn't handled in the backend?
If there are multiple NTP pages open (very common to have many) then this will get set multiple times, plus any open rewards settings page and the extension. In fact, won't the handler in the extension background script always perform the action regardless of what caused the wallet to be initialized?